src/lib/operators/table_scan/column_vs_value_table_scan_impl.cpp

```diff
@@ -113,14 +113,15 @@ void ColumnVsValueTableScanImpl::_scan_dictionary_segment(const BaseDictionarySe
       return predicate_comparator(position.value(), search_value_id);
     };
     iterable.with_iterators(position_filter, [&](auto it, auto end) {
-      if (_predicate_condition == PredicateCondition::GreaterThan ||
-          _predicate_condition == PredicateCondition::GreaterThanEquals) {
-        // For GreaterThan(Equals), INVALID_VALUE_ID would compare greater than the search_value_id, even though the
-        // value is NULL. Thus, we need to check for is_null as well.
-        _scan_with_iterators<true>(comparator, it, end, chunk_id, matches);
-      } else {
-        // No need for NULL checks here, because INVALID_VALUE_ID is always greater.
+      // INVALID_VALUE_ID (i.e. MAX_UINT) represents a NULL in the AttributeVector. For some PredicateConditions, we can
```

Comment:
My understanding so far was this: `INVALID_VALUE_ID` should only be used to represent "not found in dictionary". NULL should be represented by `dictionary.size()`.

Reply:
yes.